### ISSUE TITLE
`SHELL` builtin rule do not respect system %PATH% on Windows

### ISSUE THREAD

**Issue by Grantim** [bfgroup/b2#48]
### Environment and version details

* Operating System+version: Windows 10 x64
* Compiler+version: MSVC v142
* B2 Version: B2 Version 4.3. OS=NT
* B2 Configuration: 
```
notice: found boost-build.jam at C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/boost-build.jam
notice: loading B2 from C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel/bootstrap.jam
notice: Searching 'C:\WINDOWS' 'C:\Users\G.Karapetyan.ADALISK' 'C:\Users\G.Karapetyan.ADALISK' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/util' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/tools' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/contrib' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/.' for site-config configuration file 'site-config.jam'.
notice: Configuration file 'site-config.jam' not found in 'C:\WINDOWS' 'C:\Users\G.Karapetyan.ADALISK' 'C:\Users\G.Karapetyan.ADALISK' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/util' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/tools' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/contrib' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/.'.
notice: Searching 'C:\Users\G.Karapetyan.ADALISK' 'C:\Users\G.Karapetyan.ADALISK' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/util' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/tools' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/contrib' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/.' for user-config configuration file 'user-config.jam'.
notice: Configuration file 'user-config.jam' not found in 'C:\Users\G.Karapetyan.ADALISK' 'C:\Users\G.Karapetyan.ADALISK' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/util' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/tools' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/contrib' 'C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/.'.
Jamroot.jam:107: in modules.load from module Jamfile<C:\vcpkg2\vcpkg\installed\x64-windows-meshrus\tools\boost-build>
warning: Bison generator program 'bison' not found. Skipping grammar build.
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build\project.jam:372: in load-jamfile from module project
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build\project.jam:64: in load from module project
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/build\project.jam:142: in project.find from module project
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src\build-system.jam:618: in load from module build-system
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel\modules.jam:295: in import from module modules
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/src/kernel/bootstrap.jam:139: in boost-build from module
C:/vcpkg2/vcpkg/installed/x64-windows-meshrus/tools/boost-build/boost-build.jam:8: in module scope from module
warning: No toolsets are configured.
warning: Configuring default toolset "msvc".
warning: If the default is wrong, your build may not work correctly.
warning: Use the "toolset=xxxxx" option to override our guess.
warning: For more configuration options, please consult
warning: http://boost.org/boost-build2/doc/html/bbv2/advanced/configuration.html
notice: [msvc-cfg] msvc-14.0 detected, command: 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\cl.exe'
notice: [msvc-cfg] msvc-14.2 detected, command: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\cl.exe'
notice: [msvc-cfg] msvc-14.1 detected, command: 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x64\cl.exe'
notice: will use 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\cl.exe' for msvc, condition <toolset>msvc-14.2
notice: [generate-setup-cmd] 14.16.27023 is 14.1
notice: [generate-setup-cmd] 14.16.27023 is 14.1
notice: [generate-setup-cmd] 14.16.27023 is 14.1
notice: [generate-setup-cmd] 14.16.27023 is 14.1
notice: [generate-setup-cmd] 14.16.27023 is 14.1
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>/<address-model>', setup: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars32.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>/<address-model>32', setup: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars32.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>x86/<address-model>', setup: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars32.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>x86/<address-model>32', setup: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars32.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>/<address-model>64', setup: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsx86_amd64.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>x86/<address-model>64', setup: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsx86_amd64.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>ia64/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>ia64/<address-model>64', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>arm/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>arm/<address-model>32', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>arm/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-14.2/<architecture>arm/<address-model>64', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvarsall.bat'
...found 1 target...
```

### Brief problem description
**SHELL** builtin rule do not respect system %PATH% on Windows

### Steps to reproduce the issue
```
local rule echo-path ( )
{
    y = [ SHELL "echo %PATH% 2>&1" : exit-status ] ;
    if $(y[2]) = 0
    {
    	debug-message $(y[1]) ;
    }
    return ;
}
```

### Actual behavior summary
It outputs this
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\\Extensions\Microsoft\IntelliCode\CLI;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30037\bin\HostX64\x64;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\VCPackages;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\bin\Roslyn;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Team Tools\Performance Tools\x64;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Team Tools\Performance Tools;
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\\x64;
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\;
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\;
C:\Program Files (x86)\HTML Help Workshop;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\devinit;
C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64;
C:\Program Files (x86)\Windows Kits\10\bin\x64;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\\MSBuild\Current\Bin;
C:\WINDOWS\Microsoft.NET\Framework64\v4.0.30319;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\;
C:\vcpkg2\vcpkg\downloads\tools\powershell-core-7.1.0-windows;
C:\WINDOWS\system32;
C:\WINDOWS;
C:\WINDOWS\system32\Wbem;
C:\WINDOWS\system32\WindowsPowerShell\v1.0\;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\Llvm\x64\bin;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe 
```
### Expected behavior summary
It should output (previous block + system %PATH%)
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\\Extensions\Microsoft\IntelliCode\CLI;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30037\bin\HostX86\x86;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\VCPackages;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\bin\Roslyn;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Team Tools\Performance Tools;
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\;
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\;
C:\Program Files (x86)\HTML Help Workshop;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\devinit;
C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x86;
C:\Program Files (x86)\Windows Kits\10\bin\x86;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\\MSBuild\Current\Bin;
C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\;
C:\Program Files\Python39\Scripts\;
C:\Program Files\Python39\;
C:\Program Files\Microsoft MPI\Bin\;
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin;
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\libnvvp;
C:\Windows\system32;
C:\Windows;
C:\Windows\System32\Wbem;
C:\Windows\System32\WindowsPowerShell\v1.0\;
C:\Program Files\Microsoft DNX\Dnvm\;
C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;
C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;
C:\WINDOWS\system32;
C:\WINDOWS;
C:\WINDOWS\System32\Wbem;
C:\WINDOWS\System32\WindowsPowerShell\v1.0\;
C:\WINDOWS\System32\OpenSSH\;
C:\Program Files (x86)\OpenCTM 1.0.3\bin;
C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;
C:\Program Files\Perforce\;
C:\Program Files\CMake\bin;
C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;
C:\Program Files\PuTTY\;
C:\Qt\5.14.2\msvc2017_64\bin;
C:\Program Files (x86)\IncrediBuild;
C:\vcpkg;
C:\WINDOWS\system32;
C:\WINDOWS;
C:\WINDOWS\System32\Wbem;
C:\WINDOWS\System32\WindowsPowerShell\v1.0\;
C:\WINDOWS\System32\OpenSSH\;
C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;
C:\Program Files (x86)\GitExtensions\;
C:\Program Files\Git\cmd;
C;
C:\Program Files\Docker\Docker\resources\bin;
C:\ProgramData\DockerDesktop\version-bin;
C:\Program Files\7-Zip;
C:\PSTools;
C:\Program Files\mingw-w64\x86_64-8.1.0-win32-seh-rt_v6-rev0\mingw64\bin;
C:\Program Files (x86)\GitHub CLI\;
;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\Llvm\bin;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe
```

**Comment by grafikrobot** [comment]
I tried to reproduce this on my win10 system and I get the correct, afaict, result. I tried with a `path-echo.jam` file:

```
ECHO [ SHELL "echo %PATH%" ] ;
```

And running:
```
b2 -f path-echo.jam
```

Both running that and doing `echo %PATH%` on the `cmd.exe` console produced the same result. Can you give me more context as to how you are getting the extra content of the `PATH`?

**Comment by Grantim** [comment]
First output - is output of jam file and second is manual calling `echo %PATH%` from 
`Visual Studio 2019 Developer Command Prompt`

Thank you for helping

**Comment by grafikrobot** [comment]
If I follow these steps:

1. Launch a `Developer Command Prompt for VS 2019`
2. Run `b2 -f path_echo.jam` (which is as previous comment)
3. Run `echo %PATH`

The result is that both (2) and (3) print the same result. Can you be more specific as to the steps you followed?

**Comment by Grantim** [comment]
I faced this issue trying to build boost python with vcpkg, pls have a look at https://github.com/boostorg/build/issues/729 and https://github.com/microsoft/vcpkg/issues/5732 . The way I reproduced this issue was to rebuild boost-python with vcpkg, modifying `boost-build\src\tools\python.jam` file. 
Probably the problem is not really in `SHELL` rule and it is somewhere in vcpkg environment.
So I've got different output when build boost python (with modified python.jam file, with 
```
y = [ SHELL "echo %PATH% 2>&1" : exit-status ] ;
```
and than manually calling `echo %PATH%` )

I will close this issue, and investigate more carefully. Thank you for support!
